src/cli_util.rs

```diff
@@ -1596,6 +1600,85 @@ pub fn serialize_config_value(value: &config::Value) -> String {
     }
 }
 
+pub fn write_config_value_to_file(
+    key: &str,
+    value_str: &str,
+    path: &PathBuf,
+) -> Result<(), CommandError> {
+    // Read config
+    let config_toml = std::fs::read_to_string(path).or_else(|err| {
+        match err.kind() {
+            // If config doesn't exist yet, read as empty and we'll write one.
+            std::io::ErrorKind::NotFound => Ok("".to_string()),
+            _ => Err(user_error(format!(
+                "Failed to read file {path}: {err:?}",
+                path = path.display()
+            ))),
+        }
+    })?;
+    let mut doc = config_toml.parse::<toml_edit::Document>().map_err(|err| {
+        user_error(format!(
+            "Failed to parse file {path}: {err:?}",
+            path = path.display()
+        ))
+    })?;
+
+    // Apply config value
+    // Iterpret value as string unless it's another simple scalar type.
+    // TODO(#531): Infer types based on schema (w/ --type arg to override).
+    let item = match toml_edit::Value::from_str(value_str) {
+        Ok(value @ toml_edit::Value::Boolean(..))
+        | Ok(value @ toml_edit::Value::Integer(..))
+        | Ok(value @ toml_edit::Value::Float(..))
+        | Ok(value @ toml_edit::Value::String(..)) => toml_edit::value(value),
+        _ => toml_edit::value(value_str),
+    };
+    // TODO(#531): Prefer to create new structured tables vs inline.
+    let mut target_entry = doc.as_item_mut();
+    let mut key_parts_iter = key.split('.');
+    // Note: split guarantees at least one item.
+    let mut last_key_part = key_parts_iter.next().unwrap();
+    for key_part in &mut key_parts_iter {
+        target_entry = &mut target_entry[last_key_part];
+        last_key_part = key_part;
+    }
+    target_entry[last_key_part] = item;
```

Comment:
`item[str]` may fail if the `item` already exists and is not a table.
https://docs.rs/toml_edit/0.19.4/src/toml_edit/index.rs.html#52

I'm not sure what's the best way to look up nested tables, but maybe we'll have to use `.into_table()`, `Table::entry()`, etc.

Reply:
Thanks! Code looks a little more awkward, but it works well (and fixes one of my TODOs to prefer structured tables vs. inline).